### PR TITLE
Restructure basics of C# Directory and add example

### PR DESCRIPTION
**Pull request by asmockler**
Restructured the Basics of C# directory into 3 sub-directories:
- `example/`, which contains a sample C# application
- `tutorials/`, which contains any `.md` file that students are assigned for reading, etc.
- `assignment/`, which contains the assignment description and could optionally contain other necessary items.

I propose we have some sort of similar structure for all the sections of the class to make things a bit easier to navigate. Not sure if this is overkill, though.

Also, let me know if you think the example is too similar to the assignment.

**Important!** We will need to update links in LearningSuite if we merge this.

cc: @danmo91

### CODE REVIEW

**Comment by danmo91** [comment]
# TL;DR;

Looks GREAT! 💋 

## Details

The example program strongly demonstrates how to implement logic and shows the benefits of using `classes`, `const`,  and `Exception Handling`.

I think at this point the goal isn't to test if students can reason out an elegant solution to a problem, but to help them study and understand code that is well designed.

The [console-assignment](https://github.com/byu-is-403/syllabus/blob/master/basics-of-csharp/console-assignment.md) already provides tips for completing the project, this gives them a pattern to see how it all fits together without solving the problem for them.
